tracer/src/Datadog.Trace/Agent/AgentWriter.cs

```diff
@@ -459,6 +466,22 @@ SpanBuffer SwapBuffers()
             }
         }
 
+        private void RunSpanSampler(ArraySegment<Span> spans)
+        {
+            if (_spanSampler is null)
+            {
+                return;
+            }
+
+            foreach (var span in spans)
+            {
+                if (span.Context.TraceContext?.SamplingPriority <= 0)
```

Comment:
We might want to double-check with @lucaspimentel , but I think the SamplingPriority should be the same among all the input `spans` in this trace chunk, so we should be able to move the if statement out and enclose the foreach loop. You might need to update the check to `if (spans.Array![spans.Offset].Context.TraceContext?.SamplingPriority <= 0)`

I also think you'll want to write the for loop like this to avoid an allocation for the enumerator

```
for (int i = 0; i < spans.Count; i++)
{
    _spanSampler.MakeSamplingDecision(spans.Array[i + spans.Offset]);
}
```

Reply:
Thanks for this, I've swapped from the `foreach` to the `for` and also moved the `if` statement out.

I'll keep that in mind for the future when I do `foreach`s 👍 

fb4d3a796391e1e5c2a71beb06016c4d1aac69af